common/xbps-src/shutils/build_dependencies.sh

```diff
@@ -1,13 +1,21 @@
 # vim: set ts=4 sw=4 et:
 #
 setup_pkg_depends() {
-    local pkg="$1" out="$2" j _rpkgname _depname _pkgname foo _deps
+    local pkg="$1" out="$2" with_subpkgs="$3" j _rpkgname _depname _pkgname foo _deps collected
 
     if [[ $pkg ]]; then
         # subpkg
         if declare -f ${pkg}_package >/dev/null; then
             ${pkg}_package
         fi
+    elif [[ $with_subpkgs ]]; then
```

Comment:
hmm now that I'm looking at this more carefully, this won't work because `$pkg` is always true, so this branch will be skipped due to `elif`.